src/server.ts

```diff
@@ -1,20 +1,76 @@
 import dotenv from 'dotenv';
 dotenv.config();
 import express from 'express';
+import type { Skill } from './types';
+import {
+  getSkill,
+  readSkills,
+  addSkill,
+  deleteSkill,
+  updateSkill,
+} from './utils/skills';
+import { connectDatabase } from './utils/database';
+
+if (!process.env.MONGODB_URL) {
+  throw new Error('No MONGODB_URL dotenv variable');
+}
 
 const app = express();
 const port = process.env.PORT || 3001;
 
 app.use(express.json());
 
-app.get('/api/hello', (_request, response) => {
-  response.json({ message: 'Hello API!' });
+app.get('/api/skills', async (_request, response) => {
+  try {
+    const skills = await readSkills();
+    response.status(200).json(skills);
+  } catch (error) {
+    console.error(error);
+    response.status(500).send('Internal Server Error! Please try again later.');
+  }
+});
+
+app.get('/api/skills/:title', async (request, response) => {
+  const { title } = request.params;
+  try {
+    const skill = await getSkill(title);
+    response.status(200).json(skill);
+  } catch (error) {
+    console.error(error);
+    response.status(404).send(`Could not find skill: ${title}`);
+  }
+});
+
+app.post('/api/skills', async (request, response) => {
+  const skill: Skill = request.body;
+  const skillId = await addSkill(skill);
+  return response.status(200).send(skillId);
+});
+
+app.delete('/api/skills/:title', async (request, response) => {
+  const { title } = request.params;
+  await deleteSkill(title);
+  response.status(200).send();
+});
+
+app.patch('/api/skills/:title', async (request, response) => {
+  const { title } = request.params;
+  const skill: Skill = request.body;
+  try {
+    await updateSkill(title, skill);
+    response.status(200).json(skill);
+  } catch (error) {
+    console.error(error);
+    response.status(404).send(`Could not find skill: ${title}`);
+  }
 });
```

Comment:
Great code! In case you use many routes, consider splitting them up in several files as in this example: 
[https://github.com/marwinburesch/moviescout/blob/main/src/lib/routes/movie.ts](url)
[https://github.com/marwinburesch/moviescout/blob/6dbae1cc80f17ea1ac28af0d9288ccf4b65026f6/src/server.ts#L12](url)

That's just a suggestion and on a higher level than what you've done in the course so far ;)